crates/cast/src/cmd/creation_code.rs

```diff
@@ -131,11 +129,16 @@ pub async fn parse_code_output(
 }
 
 /// Fetches the creation code of a contract from Etherscan and RPC.
-pub async fn fetch_creation_code(
+pub async fn fetch_creation_code_from_etherscan(
     contract: Address,
-    client: Client,
+    etherscan: &EtherscanOpts,
     provider: RetryProvider,
 ) -> Result<Bytes> {
+    let config = etherscan.load_config()?;
+    let chain = config.chain.unwrap_or_default();
+    let api_version = config.get_etherscan_api_version(Some(chain));
+    let api_key = config.get_etherscan_api_key(Some(chain)).unwrap_or_default();
```

Comment:
it's funny how when you try to unify things you now suddenly need to handle another case -.-